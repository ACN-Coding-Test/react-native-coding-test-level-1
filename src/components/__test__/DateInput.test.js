import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, cleanup, fireEvent} from 'react-native-testing-library';
import {DateInput} from '..';

afterEach(cleanup);

jest.mock('@react-native-community/datetimepicker', () => jest.fn());

describe('testing custom input', () => {
  const wrapper = renderer.create(<DateInput value={new Date()} />);

  it('Should render DateInput', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  //   it.only('should fire onChange with new date', async () => {
  //     const onChange = jest.fn();

  //     const rendered = render(
  //       <DateInput onChange={onChange} value={new Date()} />,
  //     );

  //     const dateInputComponent = rendered.queryAllByTestId('dateInput');

  //     // console.log('dateInput', dateInputComponent);
  //     fireEvent(dateInputComponent, 'change', new Date());

  //     // expect(onChange).toHaveBeenCalledWith(new Date());
  //   });
});
