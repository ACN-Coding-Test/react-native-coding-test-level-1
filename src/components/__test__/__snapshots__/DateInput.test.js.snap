// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing custom input Should render DateInput 1`] = `
<View
  style={
    [
      {
        "alignSelf": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "90%",
      },
    ]
  }
  testID="dateInputParent"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderRadius": 5,
        "borderWidth": 1,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              {
                "letterSpacing": 1.5,
              },
            ],
          ],
        ]
      }
    >
      BirthDay :
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
                "letterSpacing": 0.5,
                "lineHeight": 24,
                "marginVertical": 2,
              },
              {
                "letterSpacing": 1.5,
              },
            ],
          ],
        ]
      }
    >
      Sat Nov 12 2022
    </Text>
  </View>
</View>
`;
